frontend/src/metabase/nav/components/WhatsNewNotification/WhatsNewNotification.tsx

```diff
@@ -0,0 +1,92 @@
+import { Paper } from "@mantine/core";
+import { t } from "ttag";
+import { useCallback, useMemo } from "react";
+import { updateSetting } from "metabase/admin/settings/settings";
+import IconButtonWrapper from "metabase/components/IconButtonWrapper";
+import { Icon } from "metabase/core/components/Icon";
+import { isNotFalsy } from "metabase/core/utils/types";
+import { color } from "metabase/lib/colors";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { getIsEmbedded } from "metabase/selectors/embed";
+import {
+  getLastAcknowledgedVersion,
+  getSetting,
+} from "metabase/selectors/settings";
+import { Flex, Stack, Text } from "metabase/ui";
+import { getLatestEligibleReleaseNotes } from "./utils";
+import Sparkles from "./sparkles.svg?component";
+
+export function WhatsNewNotification() {
+  const dispatch = useDispatch();
+  const isEmbedded = useSelector(getIsEmbedded);
+  const versionInfo = useSelector(state => getSetting(state, "version-info"));
+  const currentVersion = useSelector(state => getSetting(state, "version"));
+  const lastAcknowledgedVersion = useSelector(getLastAcknowledgedVersion);
+
+  const url: string | undefined = useMemo(() => {
+    if (isEmbedded || currentVersion.tag === undefined) {
```

Comment:
Could we move all this logic to the utility function so we can test if all there?